### DISCUSSION TITLE
"type_error" calling "value"-method with array at first level

### DISCUSSION THREAD

**Discussion by JHerrKodo** [nlohmann/json#3755] [Q&A] [answered]
With following example I get a type_error when calling "value" -method for array at first level. 
"at"-method works fine
"value"-method always fails

```
  json j = json::parse( "[{\"idx\":1,\"str\":\"one\"},{\"idx\":4,\"str\":\"four\"},{\"idx\":2,\"str\":\"two\"}]" ); 
  json::json_pointer p( "/0/str" ); 
  std::cout << j[p] << "\n\n"; // works fine
  std::cout << j.value(p, "fail") << "\n\n"; // type_error thrown
```

Is this an issue, or is "value" not supported for arrays?

**Comment by falbrechtskirchinger** [comment, marked as answer]
Using `value()` with arrays is not supported ... for no good reason (that I can think of). I assume that when the `json_pointer` overload was added, the code was copy&pasted without accounting for the difference.

**Reply by falbrechtskirchinger** [reply]
FYI, the documentation states:
> The function is basically equivalent to executing
> ```cpp
> try {
>    return at(ptr);
> } catch(out_of_range) {
>    return default_value;
> }
> ````

This is not the case, so this is definitely a bug and I'm working on a PR.
